src/tests/Common/Directory.Build.targets

```diff
@@ -42,7 +42,7 @@
     <MSBuild Condition="'$(Crossgen2Supported)' == 'true'"
       Targets="Restore;PublishToDisk"
       BuildInParallel="true"
-      Properties="NativeAotSupported=$(NativeAotSupported);OutputPath=$(CORE_ROOT)\crossgen2"
+      Properties="NativeAotSupported=$(NativeAotSupported);OutputPath=$(CORE_ROOT)\crossgen2;MSBuildRestoreSessionId=$([System.Guid]::NewGuid());"
```

Comment:
Assuming `MSBuildRestoreSessionId` applies only to restore, will split make any difference?

Reply:
Yes, as it will force a re-evaluation of the MSBuild project for the `PublishToDisk` target as the global properties differ from the `Restore` target invocation.